projboard/tests/test_client.py

```diff
@@ -20,15 +21,35 @@ def test_homepage(client, articles_num):
     assert 'landing/homepage.html' in template_names
 
 
-def test_get_searchpage(client):
-    response = client.get("/search/")
+@pytest.fixture
+@pytest.mark.django_db
+def User2(client):
+    return User.get_user_by_nickname("User2")
+
+
+@pytest.mark.django_db
+def test_get_searchpage(client, User2):
+    response = client.get(f"/search/{User2.nickname}/")
     assert response.status_code == 200
+    assert response.context['user'] == User2
     assert [] == response.context[ARTICLES]
     assert '' == response.context[SEARCHTITLE]
     assert 0 == response.context[COUNT]
     assert '' == response.context[MESSAGE]
 
 
+@pytest.mark.django_db
+def test_get_empty_searchpage(client):
+    response = client.get("/search/")
+    assert response.status_code == 404
+
+
+@pytest.mark.django_db
+def test_get_ivalid_nickname_searchpage(client, User2):
```

Comment:
this fixture is not relevant here 
since you are format the string to fakenickname

Reply:
done